### PR TITLE
Essential updates and some improvements to the release script and Vagrantfile

### PR DESCRIPTION
**Pull request by rdowner**

### CODE REVIEW

**Comment by ahgittin** [comment]
I've just eyeballed the changes, thinking it's better to have these in than not, and it is still evolving quite rapidly.

We'll be testing vagrant and build in some detail soon.  @johnmccabe you may want to cast an eye over the `Vagrantfile` changes to make sure they're sensible to you.

(merging, we can improve with a follow-up PR if needed)

**Comment by rdowner** [comment]
@johnmccabe @ahgittin This Vagrantfile is for the release build environment - it's not John's new Vagrantfile for end users.

**Comment by johnmccabe** [comment]
@ahgittin @rdowner looking at the Vagrant stuff (not the build env) is on my radar, I wanted to try getting a test blueprint running using Softlayer Bare-Metal (it doesn't like nested virtualisation).

File: release/clone-and-configure-repos.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+set -e
+set -x
+
+# Do the basics
+git clone http://github.com/apache/brooklyn/
```
Comment (neykov) [inline release/clone-and-configure-repos.sh]: Why not clone from apache directly?

Comment (richardcloudsoft) [inline release/clone-and-configure-repos.sh]: :flushed: 

I honestly cannot give an answer for why I wrote the script to checkout from GitHub, then delete the GitHub remote, then add the Apache one (other than I copy-and-pasted from the website "Get the code" instructions) :confused:

**Comment by richardcloudsoft** [comment]
BTW @ahgittin commit 2f19749 changes something that you had added previously - wanted to draw your attention to it so you have the opportunity to object!

**Comment by ahgittin** [comment]
thanks for checking @richardcloudsoft -- i've no issues with svn not committing automatically.  presumably the instructions tell you what you need to commit.

**Pull request merged**